Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1

```diff
@@ -31,6 +31,35 @@ function Format-Splat {
     "$parameters" # String join the array.
 }
 
+function Generate-TlsError {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)] $exception 
+    )
+
+    if ($exception -eq $null)
+    {
+        return
+    }
+
+    $isWebException = $true
+    $innerException = $exception
+    while ($innerException.GetType() -ne [System.Net.WebException])
+    {
+        $innerException = $innerException.InnerException
+        if ($innerException -eq $null)
+        {
+            $isWebException = $false
+            break
+        }
+    }
+
+    if (($isWebException -eq $true) -and ($innerException.InnerException -ne $null) -and ($innerException.InnerException.GetType() -eq [System.IO.IOException]))
```

Comment:
why InnerException of $innerException?

Reply:
We are looking for WebException and IOException. When these 2 are thrown, then there is a high chance it is due to TLS

Reply:
here $innerException is child of exception of type WebException. That means you are looking for a scenario where its grand-child is exception of type IOException.
Is this really going to work for all cases? Have you tested it with anything other than login azure RM?
Should we be bit more dumb and always print this message(mentioned it as one probable cause) when exception happens